### DISCUSSION TITLE
Leaf Sequencing data

### DISCUSSION THREAD

**Discussion by naninano1** [e0404/matRad#539] [Q&A]
Hello,

I am performing my second project with matRad. I like it.

Now, I want to transfer the leaf sequences of a specific IMRT plan of matRad into TOPAS to compare the dosimetric results.
I have two questions:
1-In GUI I can enable the sequencing and then see the leaves arrangments after optimization. However, in the code based running when I enable pln.propOpt.runSequencing   = true, after optimization nothing is shown in regard to the leaf arrangments.
Moreover, I tried to find the leaf sequence matrix or data as mentioned by @wahln in a topic (https://github.com/e0404/matRad/discussions/523#discussioncomment-1207275), but I couldn't find anything.

2- Is there a way to compare the dose maps in terms of gamma index (or DVHs) directly in matRad, similar to the plan comparison which can be done by matRad_compareDose ?

Thanks

**Comment by naninano1** [comment]
Dear Niklas,

I would really appreciate it if you would guide me in this regard.
@wahln

**Reply by wahln** [reply]
ah sorry, somehow this went unnoted. 

1) The sequencing needs to be separately called. this is bad design at the moment. Check out the matRad.m script -> after optimization, you can call one of the sequencers manually, e.g. with `resultGUI = matRad_xiaLeafSequencing(resultGUI,stf,dij,numOfLevels)`
2) You can use the `matRad_gammaIndex` function directly, which takes two cubes and a few parameters as input. For comparing DVHs you can call `matRad_calcDVH` directly and then plot them manually, which takes a dose cube, the cst, the dosegrid etc. as parameters. just check out the header definition in both cases.

**Reply by naninano1** [reply]
Dear Niklas,

Thank you for your reply. I am going to try the mentioned methods.

**Reply by naninano1** [reply]
Hello Niklas,

I checked the mentioned comment in part 1. I think when we enable: `pln.propOpt.runSequencing   = true;` the mentioned comment will be automatically enabled in this section:
`if strcmp(pln.radiationMode,'photons') && (pln.propOpt.runSequencing || pln.propOpt.runDAO)
    %resultGUI = matRad_xiaLeafSequencing(resultGUI,stf,dij,5);
    %resultGUI = matRad_engelLeafSequencing(resultGUI,stf,dij,5);
    resultGUI = matRad_siochiLeafSequencing(resultGUI,stf,dij,1);
end`

However, I called the sequence visualization through this commend: `matRad_visApertureInfo(resultGUI.apertureInfo);`

**Reply by wahln** [reply]
yes, thanks for reporting this. 
Still, it is bad design from us since this if statement is only used in the matRadGUI or in scripts. Thus, when you set the parameter before optimization, nothing happens. Sequencing only happens when it is called explicitly, but if it is a propOpt parameter, people might expect that sequencing is run explicitly in the call to the optimization routine.
